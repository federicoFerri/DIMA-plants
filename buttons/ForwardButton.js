import React from 'react';
import { StyleSheet, Image, TouchableOpacity, SafeAreaView } from 'react-native';

const image_ForwardButton = require('../assets/button_images/arrow_forward.png');

{/*props required
  @onPress: if button is pressed, it launches an event like TouchableOpacity
*/}

class ForwardButton extends React.Component {
    imagePressed = () => () => {
        this.props.onPress()
    }
    render() {
        return(
            <SafeAreaView>
                <TouchableOpacity 
                    activeOpacity={0.5}
                    onPress={this.imagePressed()}
                >
                    <Image 
                        style={stylesImage.backButton}
                        source={image_ForwardButton}
                        /> 
                </TouchableOpacity>
            </SafeAreaView>
        )
    }
  
}


const stylesImage = StyleSheet.create({
    flex: 1,
    forwardButton: {
      width: 28,
      height: 23,
    },
  });

export default ForwardButton;