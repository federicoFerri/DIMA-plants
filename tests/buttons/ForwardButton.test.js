import React from 'react';
import renderer from 'react-test-renderer';

import ForwardButton from '../../buttons/ForwardButton'

describe('<ForwardButton />', () => {
    test('renders correctly', () => {
        const tree = renderer.create(<ForwardButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});