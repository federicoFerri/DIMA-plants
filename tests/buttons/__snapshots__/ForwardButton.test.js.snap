// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForwardButton /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 27,
          "resizeMode": "contain",
          "width": 32,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
